File: app/src/main/java/org/schabi/newpipe/fragments/list/search/SuggestionItem.java
package org.schabi.newpipe.fragments.list.search;

import androidx.annotation.NonNull;

public class SuggestionItem {
    final boolean fromHistory;
    public final String query;

    public SuggestionItem(final boolean fromHistory, final String query) {
        this.fromHistory = fromHistory;
        this.query = query;
    }

    @Override
    public boolean equals(final Object o) {
        if (o instanceof SuggestionItem) {
            return query.equals(((SuggestionItem) o).query);
        }
        return false;
    }

    @Override
    public int hashCode() {
        return query.hashCode();
    }

    @NonNull
    @Override
    public String toString() {
        return "[" + fromHistory + "→" + query + "]";
    }
}

File: app/src/main/java/org/schabi/newpipe/fragments/list/search/SuggestionItem.kt
/*
 * SPDX-FileCopyrightText: 2017-2025 NewPipe contributors <https://newpipe.net>
 * SPDX-License-Identifier: GPL-3.0-or-later
 */

package org.schabi.newpipe.fragments.list.search

class SuggestionItem(@JvmField val fromHistory: Boolean, @JvmField val query: String) {
    override fun equals(other: Any?): Boolean {
        if (other is SuggestionItem) {
            return query == other.query
        }
        return false
    }

    override fun hashCode() = query.hashCode()

    override fun toString() = "[$fromHistory→$query]"
}
